File: benchmarks/execution/Java_Clojure_Set_Big_Integer/TrafficLightExec.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class TrafficLightExec {


    private TrafficLight _TrafficLight = new TrafficLight();



    private BInteger counter;

    public TrafficLightExec() {
        counter = new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("300000000"))).booleanValue()) {
            this._TrafficLight.cars_ry();
            this._TrafficLight.cars_g();
            this._TrafficLight.cars_y();
            this._TrafficLight.cars_r();
            this._TrafficLight.peds_g();
            this._TrafficLight.peds_r();
            counter = counter.plus(new BInteger("1"));
        }
    }

    public BInteger getCounter() {
        BInteger out = null;
        out = counter;
        return out;
    }

    public static void main(String[] args) {
        TrafficLightExec exec = new TrafficLightExec();
        exec.simulate();
    }

}

File: benchmarks/execution/Rust_Primitives/TrafficLightExec.rs
#![ allow( dead_code, unused, non_snake_case, non_camel_case_types, unused_assignments ) ]
use std::fmt;
use rand::{thread_rng, Rng};
use btypes::butils;
use btypes::bobject;
use btypes::bboolean::{IntoBool, BBooleanT};
use btypes::binteger::BInteger;
use btypes::bboolean::BBoolean;
mod TrafficLight;



#[derive(Clone, Default, Debug, Hash, PartialEq, Eq)]
pub struct TrafficLightExec {
    counter: BInteger,
    _TrafficLight: TrafficLight::TrafficLight,
}

impl fmt::Display for TrafficLightExec {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let mut result = "TrafficLightExec: (".to_owned();
        result += &format!("_get_counter: {}, ", self._get_counter());
        result = result + ")";
        return write!(f, "{}", result);
    }
}

impl TrafficLightExec {

    pub fn new() -> TrafficLightExec {
        //values: ''
        let mut m: TrafficLightExec = Default::default();
        m.init();
        return m;
    }
    fn init(&mut self) {
        self.counter = BInteger::new(0);
        self._TrafficLight = TrafficLight::TrafficLight::new();
    }

    pub fn _get_counter(&self) -> BInteger {
        return self.counter.clone();
    }

    pub fn simulate(&mut self) -> () {
        while (self.counter.less(&BInteger::new(300000000))).booleanValue() {
            self._TrafficLight.cars_ry();
            self._TrafficLight.cars_g();
            self._TrafficLight.cars_y();
            self._TrafficLight.cars_r();
            self._TrafficLight.peds_g();
            self._TrafficLight.peds_r();
            self.counter = self.counter.plus(&BInteger::new(1));
        }
    }
}

fn main() {
    let mut exec = TrafficLightExec::new();
    exec.simulate();
}
